gpkit/small_classes.py

```diff
@@ -247,8 +241,9 @@ def __add__(self, other):
             return self.__class__({key: val+other
                                    for (key, val) in self.items()})
         elif isinstance(other, dict):
-            keys = set(self).union(other)
-            sums = {key: self.get(key, 0) + other.get(key, 0) for key in keys}
+            sums = self.copy()
+            for key, value in other.items():
+                sums[key] = value + sums.get(key, 0)
             return self.__class__(sums)
```

Comment:
Then wouldn't it be cleaner to implement `copy()` for this class? It calls super and does the cast.